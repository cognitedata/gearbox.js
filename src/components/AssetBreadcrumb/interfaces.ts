import { Asset, CogniteInternalId } from '@cognite/sdk';

export type FetchAssetCall = (assetId: CogniteInternalId) => Promise<Asset>;

export interface AssetBreadcrumbProps {
  /**
   * Asset ID
   */
  assetId: number;
  /**
   * Maximal number of assets to be displayed.
   * If length of the asset chain bigger than < maxLength > value,
   * asset chain will be shrunk to root element plus < maxLength > - 1 number
   * of last elements in a chain
   */
  maxLength?: number;
  /**
   * Function, that can be used for custom rendering of asset in a breadcrumb
   */
  renderItem?: (asset: Asset, depth: number) => JSX.Element;
  /**
   * Function that can be used to replace embedded asset fetching
   */
  retrieveAsset?: FetchAssetCall;
  /**
   * Callback which is executed on a click action on asset in a breadcrumb.
   * It's only available in case of default asset rendering in a breadcrumb
   */
  onBreadcrumbClick?: (asset: Asset, depth: number) => void;
}
