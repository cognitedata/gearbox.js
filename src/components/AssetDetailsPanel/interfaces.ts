import { CSSProperties } from 'react';
import { WithAssetDataProps, WithAssetProps } from '../../hoc';
import { MetaDescriptionListProps } from '../DescriptionList';

export interface AssetDetailsPanelStylesProps {
  /**
   * Object that defines inline CSS style for container of the table
   */
  styles?: CSSProperties;
}

export type AssetDetailsPanelPureProps = WithAssetDataProps &
  MetaDescriptionListProps &
  AssetDetailsPanelStylesProps;

export type AssetDetailsPanelProps = WithAssetProps &
  MetaDescriptionListProps &
  AssetDetailsPanelStylesProps;
