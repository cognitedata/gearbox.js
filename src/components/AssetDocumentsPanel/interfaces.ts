import { FilesMetadata } from '@cognite/sdk';
import { CollapseProps } from 'antd/lib/collapse';
import { CSSProperties, ReactNode } from 'react';
import { WithAssetFilesDataProps, WithAssetFilesProps } from '../../hoc';
import { AnyIfEmpty } from '../../interfaces';

export type Document = FilesMetadata;

export type DocumentRenderer = (
  document: Document,
  i: number,
  documents: Document[]
) => ReactNode;

export type OnDocumentClick = (
  document: Document,
  category: string,
  description: string
) => void;

export interface JsonDocTypes {
  [s: string]: string;
}

export interface DocumentType {
  key: string;
  description: string;
}

export interface DocumentsByCategory {
  [s: string]: {
    description: string;
    documents: Document[];
  };
}

export interface Priority {
  [s: string]: number;
}

export interface DocumentsPanelStylesProps {
  /**
   * Object that defines inline CSS styles for inner elements of the component
   */
  styles?: AssetDocumentsPanelStyles;
}

export interface DocumentsPanelThemeProps {
  theme?: AnyIfEmpty<{}>;
}

export type DocumentTableProps = MetaDocProps &
  WithAssetFilesDataProps &
  DocumentsPanelStylesProps &
  DocumentsPanelThemeProps;

export interface AssetDocumentsPanelStyles {
  wrapper?: CSSProperties;
  fileContainer?: CSSProperties;
  fileLink?: CSSProperties;
  fileTitle?: CSSProperties;
}

export interface MetaDocProps {
  /**
   * Callback function triggered when user clicks on a file (document)
   */
  handleDocumentClick?: OnDocumentClick;
  /**
   * Object with props to be passed to atn design Collapse component
   */
  collapseProps?: CollapseProps;
  /**
   * List of categories codes to be shown on the top of the list.
   * Categories "P&ID" and "Logic Diagrams" are prioritized by default.
   */
  categoryPriorityList?: string[];
  /**
   * Name for the category that includes all files with undefined category
   */
  unknownCategoryName?: string;
  /**
   * Key to get document title from `metadata` object
   */
  documentTitleField?: string;
  /**
   * The `metadata` field used to group documents into types. If this is
   * not specified, then `doc_type` field will be attempted. If no
   * `doc_type` field is present in the metadata, then the filename will
   * be attempted to be parsed using the NORSOK standard. If this fails, then
   * the document will be in the "unknown" group.
   */
  documentTypeField?: string;
  /**
   * Object map used to show custom category names
   */
  docTypes?: JsonDocTypes;
  /**
   * Text to be shown if no documents have been found for the asset
   */
  noDocumentsSign?: string;
  /**
   * Custom render function to modify documents appearing
   */
  documentRenderer?: DocumentRenderer;
  /**
   * Function to sort categories after theirs prioritization
   */
  customCategorySort?: (a: string, b: string) => number;
}

export type AssetDocumentsPanelProps = WithAssetFilesProps &
  MetaDocProps &
  DocumentsPanelStylesProps;
