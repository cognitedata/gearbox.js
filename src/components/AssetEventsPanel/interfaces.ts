import { CSSProperties } from 'react';
import { WithAssetEventsDataProps, WithAssetEventsProps } from '../../hoc';
import { AnyIfEmpty } from '../../interfaces';

export interface TableColumnType {
  title: string;
  dataIndex: string;
  key?: string;
}

export interface TableDesignType {
  /**
   * Configure table pagination appearance
   */
  pagination?: {
    pageSize?: number;
    position?: 'top' | 'bottom' | 'both';
    showSizeChanger?: boolean;
  };
  /**
   * Configure scroll appearance
   */
  scroll?: {
    y?: string;
    x?: string;
  };
  /**
   * Define if table has borders
   */
  bordered?: boolean;
  /**
   * Toggles table header visibility
   */
  showHeader?: boolean;
  /**
   * Custom table styles
   */
  style?: CSSProperties;
}

export interface AssetEventsPanelStyles {
  table?: CSSProperties;
  tableRow?: CSSProperties;
  tableCell?: CSSProperties;
}

export interface MetaEventsProps extends TableDesignType {
  /**
   * Array of objects that customize titles of the columns in the table
   */
  columns?: TableColumnType[];
}

export interface AssetEventsPanelStylesProps {
  /**
   * Object that defines inline CSS styles for inner elements of the component
   */
  styles?: AssetEventsPanelStyles;
}

export interface AssetEventsPanelThemeProps {
  /**
   * @ignore
   */
  theme?: AnyIfEmpty<{}>;
}

export type AssetEventsPanelProps = WithAssetEventsProps &
  MetaEventsProps &
  AssetEventsPanelStylesProps &
  AssetEventsPanelThemeProps;

export type AssetEventsPanelPropsPure = WithAssetEventsDataProps &
  MetaEventsProps &
  AssetEventsPanelStylesProps &
  AssetEventsPanelThemeProps;
