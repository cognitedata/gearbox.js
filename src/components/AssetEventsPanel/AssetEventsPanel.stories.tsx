import React from 'react';
import { storiesOf } from '@storybook/react';
import AssetEventsPanel from 'components/AssetEventsPanel/AssetEventsPanel';
import { EVENTS } from 'mocks/events';

storiesOf('AssetEventsPanel', module)
  .addParameters({
    info: {
      inline: true,
      maxPropObjectKeys: 10,
    },
  })
  .add('Minimal', () => <AssetEventsPanel events={EVENTS} />)
  .add('Empty table', () => <AssetEventsPanel events={[]} />)
  .add('Pagination on top', () => (
    <AssetEventsPanel events={EVENTS} pagination={{ position: 'top' }} />
  ))
  .add('Pagination top and bottom', () => (
    <AssetEventsPanel events={EVENTS} pagination={{ position: 'both' }} />
  ))
  .add('Items per page', () => (
    <AssetEventsPanel events={EVENTS} pagination={{ pageSize: 2 }} />
  ))
  .add('Let user select pagesize', () => (
    <AssetEventsPanel events={EVENTS} pagination={{ showSizeChanger: true }} />
  ))
  .add('Hide headers', () => (
    <AssetEventsPanel events={EVENTS} showHeader={false} />
  ))
  .add('Borders in table', () => (
    <AssetEventsPanel events={EVENTS} bordered={true} />
  ))
  .add('Custom styling', () => (
    <AssetEventsPanel
      events={EVENTS}
      style={{
        border: '2px dashed red',
        padding: '0 6px',
      }}
    />
  ))
  .add('Define other columns', () => (
    <AssetEventsPanel
      events={EVENTS}
      columns={[
        {
          title: 'ID',
          dataIndex: 'id',
          key: 'id',
        },
        {
          title: 'Type',
          dataIndex: 'type',
          key: 'type',
        },
        {
          title: 'Description',
          dataIndex: 'description',
          key: 'description',
        },
        {
          title: 'Start',
          dataIndex: 'start',
          key: 'start',
        },
      ]}
    />
  ));
