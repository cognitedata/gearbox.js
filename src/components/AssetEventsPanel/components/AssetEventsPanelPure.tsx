import { CogniteEvent } from '@cognite/sdk';
import { Icon, Modal, Table } from 'antd';
import React from 'react';
import styled from 'styled-components';
import { withDefaultTheme } from '../../../hoc';
import { defaultTheme } from '../../../theme/defaultTheme';
import { momentFromTimestamp } from '../../../utils/formatters';
import { AssetEventsPanelPropsPure, EventAddonsProp } from '../interfaces';

interface AssetEventsPanelState {
  selectedEvent: EventAddonsProp | null;
}

const defaultRenderEventMetadata = (event: EventAddonsProp) => (
  <EventMetadataList key="event-metadata">
    {Object.keys(event.metadata || {}).map(key => (
      <div key={key} className="event-metadata">
        <strong>{key}</strong> <br />
        <span>{event.metadata && event.metadata[key]}</span>
      </div>
    ))}
  </EventMetadataList>
);

export class AssetEventsPanelPureComponent extends React.Component<
  AssetEventsPanelPropsPure,
  AssetEventsPanelState
> {
  static defaultProps = {
    theme: { ...defaultTheme },
  };

  constructor(props: AssetEventsPanelPropsPure) {
    super(props);
    this.state = {
      selectedEvent: null,
    };
  }

  getTableComponents() {
    const { styles } = this.props;

    return {
      body: {
        row: (props: any) => {
          return (
            <StyledRow
              style={styles && styles.tableRow}
              onClick={() => this.onEventClick(props['data-row-key'])}
            >
              {props.children}
            </StyledRow>
          );
        },
        cell: (props: any) => {
          return (
            <StyledCell style={styles && styles.tableCell}>
              {props.children}
            </StyledCell>
          );
        },
      },
    };
  }

  renderEventDetails = (event: EventAddonsProp) => (
    <EventDetails key="event-detail">
      <div key="event-type">
        <strong>{event.type},</strong> {event.subtype}
      </div>
      {event.description && (
        <div className="description">{event.description}</div>
      )}
      <div key="event-time" className="times">
        <span>{event.start}</span>
        <span>
          <Icon type="right" />
        </span>
        <span style={{ textAlign: 'right' }}>{event.end}</span>
      </div>
    </EventDetails>
  );

  mapEvent = (event: CogniteEvent): EventAddonsProp => ({
    ...event,
    typeAndSubtype: (
      <span>
        <strong style={{ display: 'block' }}>{event.type}</strong>
        <span style={{ fontSize: 12, opacity: 0.8 }}>{event.subtype}</span>
      </span>
    ),
    description: event.description || 'No description',
    start:
      typeof event.startTime === 'number'
        ? momentFromTimestamp(event.startTime).format('LLL')
        : '-',
    end:
      typeof event.endTime === 'number'
        ? momentFromTimestamp(event.endTime).format('LLL')
        : 'Ongoing',
  });

  onEventClick = (id: number) => {
    const { assetEvents } = this.props;
    if (!assetEvents) {
      return;
    }
    const selectedEvent = assetEvents.find(e => e.id === id);
    if (!selectedEvent) {
      return;
    }
    this.setState({
      selectedEvent: this.mapEvent(selectedEvent),
    });
  };

  render() {
    const defaultColumns = [
      {
        title: 'Type / Subtype',
        dataIndex: 'typeAndSubtype',
        key: 'typeAndSubtype',
      },
      {
        title: 'Description',
        dataIndex: 'description',
        key: 'description',
      },
      {
        title: 'Start',
        dataIndex: 'start',
        key: 'start',
      },
      {
        title: 'End',
        dataIndex: 'end',
        key: 'end',
      },
    ];

    const {
      columns = defaultColumns,
      assetEvents = [],
      pagination = { pageSize: 12 },
      scroll,
      bordered = false,
      showHeader = true,
      styles,
    } = this.props;

    const { selectedEvent } = this.state;
    const { renderEventMetadata = defaultRenderEventMetadata } = this.props;

    return (
      <>
        <StyledTable
          key="styleTable"
          columns={columns}
          dataSource={assetEvents.map(this.mapEvent)}
          rowKey="id"
          showHeader={showHeader}
          pagination={pagination}
          scroll={scroll}
          bordered={bordered}
          components={this.getTableComponents()}
          style={styles && styles.table}
        />
        {!!selectedEvent && (
          <Modal
            key="Modal"
            visible={!!selectedEvent}
            footer={null}
            onCancel={() => this.setState({ selectedEvent: null })}
          >
            {!!selectedEvent && [
              this.renderEventDetails(selectedEvent),
              renderEventMetadata(selectedEvent),
            ]}
          </Modal>
        )}
      </>
    );
  }
}

const StyledTable = styled(Table)``;

const StyledRow = styled.tr`
  background: white;
  cursor: pointer;
  &:nth-child(2n) {
    background: #fafafa;
  }
`;

const StyledCell = styled.td`
  max-width: 500px;
`;

const EventDetails = styled.div`
  .description {
    font-size: 18px;
    padding-top: 16px;
  }
  .times {
    display: flex;
    justify-content: space-between;
    padding-top: 16px;
  }
`;

const EventMetadataList = styled.div`
  margin-top: 32px;
  max-height: calc(100vh - 400px);
  overflow: auto;
  .event-metadata {
    padding: 16px;
    &:nth-child(2n) {
      background: #fbfbfb;
    }
  }
`;

const Component = withDefaultTheme(AssetEventsPanelPureComponent);
Component.displayName = 'AssetEventsPanelPure';

export { Component as AssetEventsPanelPure };
