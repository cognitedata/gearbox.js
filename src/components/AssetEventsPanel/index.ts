export * from './AssetEventsPanel';
export * from './interfaces';
