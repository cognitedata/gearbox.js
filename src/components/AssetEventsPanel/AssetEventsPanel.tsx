import { withAssetEvents } from '../../hoc';
import { AssetEventsPanelPure } from './components/AssetEventsPanelPure';

export const AssetEventsPanel = withAssetEvents(AssetEventsPanelPure);
