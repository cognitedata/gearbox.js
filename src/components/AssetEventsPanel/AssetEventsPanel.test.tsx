import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { fakeEvents, MockCogniteClient } from '../../mocks';
import { ClientSDKProvider } from '../ClientSDKProvider';
import { LoadingBlock } from '../common/LoadingBlock/LoadingBlock';
import { AssetEventsPanel } from './AssetEventsPanel';
import { AssetEventsPanelPure } from './components/AssetEventsPanelPure';
import { AssetEventsPanelProps } from './interfaces';

configure({ adapter: new Adapter() });

const mockEventsList = jest.fn();

class CogniteClient extends MockCogniteClient {
  events: any = {
    list: mockEventsList,
  };
}

const sdk = new CogniteClient({ appId: 'gearbox test' });

describe('AssetEventsPanel', () => {
  beforeEach(() => {
    mockEventsList.mockReturnValue({
      autoPagingToArray: () => Promise.resolve(fakeEvents),
    });
  });

  it('Should render without exploding and load data', done => {
    const props: AssetEventsPanelProps = { assetId: 123 };
    const wrapper = mount(
      <ClientSDKProvider client={sdk}>
        <AssetEventsPanel {...props} />
      </ClientSDKProvider>
    );
    expect(wrapper.find(LoadingBlock)).toHaveLength(1);

    setImmediate(() => {
      wrapper.update();
      const pureComponent = wrapper.find(AssetEventsPanelPure);
      expect(pureComponent).toHaveLength(1);
      expect(pureComponent.props().assetEvents).toEqual(fakeEvents);
      done();
    });
  });
});
