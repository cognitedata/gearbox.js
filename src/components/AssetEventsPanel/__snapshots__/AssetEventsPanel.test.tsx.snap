// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssetEventsPanel renders correctly 1`] = `
<div
  className="ant-table-wrapper sc-bdVaJa eXrEpW"
>
  <div
    className="ant-spin-nested-loading"
  >
    <div
      className="ant-spin-container"
    >
      <div
        className="ant-table ant-table-default ant-table-scroll-position-left"
        style={Object {}}
      >
        <div
          className="ant-table-content"
        >
          <div
            className="ant-table-body"
            onScroll={[Function]}
            onWheel={[Function]}
            style={Object {}}
          >
            <table
              className=""
              style={Object {}}
            >
              <colgroup>
                <col
                  style={
                    Object {
                      "minWidth": undefined,
                      "width": undefined,
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "minWidth": undefined,
                      "width": undefined,
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "minWidth": undefined,
                      "width": undefined,
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "minWidth": undefined,
                      "width": undefined,
                    }
                  }
                />
              </colgroup>
              <thead
                className="ant-table-thead"
              >
                <tr
                  style={
                    Object {
                      "height": null,
                    }
                  }
                >
                  <th
                    className=""
                  >
                    <div>
                      Type / Subtype
                    </div>
                  </th>
                  <th
                    className=""
                  >
                    <div>
                      Description
                    </div>
                  </th>
                  <th
                    className=""
                  >
                    <div>
                      Start
                    </div>
                  </th>
                  <th
                    className=""
                  >
                    <div>
                      End
                    </div>
                  </th>
                </tr>
              </thead>
              <tbody
                className="ant-table-tbody"
              >
                <tr
                  className="ant-table-row ant-table-row-level-0"
                  data-row-key={1995162693488}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onDoubleClick={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  style={
                    Object {
                      "height": null,
                    }
                  }
                >
                  <td
                    className=""
                    onClick={[Function]}
                  >
                    <span
                      className="ant-table-row-indent indent-level-0"
                      style={
                        Object {
                          "paddingLeft": "0px",
                        }
                      }
                    />
                    <span>
                      <strong
                        style={
                          Object {
                            "display": "block",
                          }
                        }
                      >
                        failure
                      </strong>
                      <span
                        style={
                          Object {
                            "fontSize": 12,
                            "opacity": 0.8,
                          }
                        }
                      >
                        Valhall
                      </span>
                    </span>
                  </td>
                  <td
                    className=""
                    onClick={[Function]}
                  >
                    Lorem ipsum dolor sit amet, consectetur adipiscing elit.
                  </td>
                  <td
                    className=""
                    onClick={[Function]}
                  >
                    April 24, 2019 3:35 PM
                  </td>
                  <td
                    className=""
                    onClick={[Function]}
                  >
                    Ongoing
                  </td>
                </tr>
                <tr
                  className="ant-table-row ant-table-row-level-0"
                  data-row-key={8825861064387}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onDoubleClick={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  style={
                    Object {
                      "height": null,
                    }
                  }
                >
                  <td
                    className=""
                    onClick={[Function]}
                  >
                    <span
                      className="ant-table-row-indent indent-level-0"
                      style={
                        Object {
                          "paddingLeft": "0px",
                        }
                      }
                    />
                    <span>
                      <strong
                        style={
                          Object {
                            "display": "block",
                          }
                        }
                      >
                        alert
                      </strong>
                      <span
                        style={
                          Object {
                            "fontSize": 12,
                            "opacity": 0.8,
                          }
                        }
                      >
                        Val
                      </span>
                    </span>
                  </td>
                  <td
                    className=""
                    onClick={[Function]}
                  >
                    sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
                  </td>
                  <td
                    className=""
                    onClick={[Function]}
                  >
                    April 27, 2019 3:35 PM
                  </td>
                  <td
                    className=""
                    onClick={[Function]}
                  >
                    April 30, 2019 3:35 PM
                  </td>
                </tr>
                <tr
                  className="ant-table-row ant-table-row-level-0"
                  data-row-key={25496029326330}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onDoubleClick={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  style={
                    Object {
                      "height": null,
                    }
                  }
                >
                  <td
                    className=""
                    onClick={[Function]}
                  >
                    <span
                      className="ant-table-row-indent indent-level-0"
                      style={
                        Object {
                          "paddingLeft": "0px",
                        }
                      }
                    />
                    <span>
                      <strong
                        style={
                          Object {
                            "display": "block",
                          }
                        }
                      >
                        workorder
                      </strong>
                      <span
                        style={
                          Object {
                            "fontSize": 12,
                            "opacity": 0.8,
                          }
                        }
                      >
                        Val
                      </span>
                    </span>
                  </td>
                  <td
                    className=""
                    onClick={[Function]}
                  >
                    Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.
                  </td>
                  <td
                    className=""
                    onClick={[Function]}
                  >
                    May 7, 2019 3:35 PM
                  </td>
                  <td
                    className=""
                    onClick={[Function]}
                  >
                    May 14, 2019 3:35 PM
                  </td>
                </tr>
                <tr
                  className="ant-table-row ant-table-row-level-0"
                  data-row-key={33766051546406}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onDoubleClick={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  style={
                    Object {
                      "height": null,
                    }
                  }
                >
                  <td
                    className=""
                    onClick={[Function]}
                  >
                    <span
                      className="ant-table-row-indent indent-level-0"
                      style={
                        Object {
                          "paddingLeft": "0px",
                        }
                      }
                    />
                    <span>
                      <strong
                        style={
                          Object {
                            "display": "block",
                          }
                        }
                      >
                        failure
                      </strong>
                      <span
                        style={
                          Object {
                            "fontSize": 12,
                            "opacity": 0.8,
                          }
                        }
                      >
                        Val
                      </span>
                    </span>
                  </td>
                  <td
                    className=""
                    onClick={[Function]}
                  >
                    Excepteur sint occaecat cupidatat non proident
                  </td>
                  <td
                    className=""
                    onClick={[Function]}
                  >
                    April 4, 2019 3:35 PM
                  </td>
                  <td
                    className=""
                    onClick={[Function]}
                  >
                    April 14, 2019 3:35 PM
                  </td>
                </tr>
                <tr
                  className="ant-table-row ant-table-row-level-0"
                  data-row-key={35593709738144}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onDoubleClick={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  style={
                    Object {
                      "height": null,
                    }
                  }
                >
                  <td
                    className=""
                    onClick={[Function]}
                  >
                    <span
                      className="ant-table-row-indent indent-level-0"
                      style={
                        Object {
                          "paddingLeft": "0px",
                        }
                      }
                    />
                    <span>
                      <strong
                        style={
                          Object {
                            "display": "block",
                          }
                        }
                      >
                        alert
                      </strong>
                      <span
                        style={
                          Object {
                            "fontSize": 12,
                            "opacity": 0.8,
                          }
                        }
                      >
                        Val
                      </span>
                    </span>
                  </td>
                  <td
                    className=""
                    onClick={[Function]}
                  >
                    Sunt in culpa qui officia deserunt mollit anim id est laborum.
                  </td>
                  <td
                    className=""
                    onClick={[Function]}
                  >
                    June 6, 2019 3:35 PM
                  </td>
                  <td
                    className=""
                    onClick={[Function]}
                  >
                    June 13, 2019 3:35 PM
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
      <ul
        className="ant-pagination ant-table-pagination"
        style={Object {}}
        unselectable="unselectable"
      >
        <li
          aria-disabled={true}
          className="ant-pagination-disabled ant-pagination-prev"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={null}
          title="Previous Page"
        >
          <a
            className="ant-pagination-item-link"
          >
            <i
              aria-label="icon: left"
              className="anticon anticon-left"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="left"
                fill="currentColor"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 0 0 0 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                />
              </svg>
            </i>
          </a>
        </li>
        <li
          className="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex="0"
          title={1}
        >
          <a>
            1
          </a>
        </li>
        <li
          aria-disabled={true}
          className="ant-pagination-disabled ant-pagination-next"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={null}
          title="Next Page"
        >
          <a
            className="ant-pagination-item-link"
          >
            <i
              aria-label="icon: right"
              className="anticon anticon-right"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="right"
                fill="currentColor"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M765.7 486.8L314.9 134.7A7.97 7.97 0 0 0 302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 0 0 0-50.4z"
                />
              </svg>
            </i>
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
